packages/api/src/controllers/stream.ts

```diff
@@ -193,13 +195,42 @@ export function getPlaybackUrl(ingest: string, stream: DBStream) {
   return pathJoin(ingest, `hls`, stream.playbackId, `index.m3u8`);
 }
 
-function getRecordingUrl(ingest: string, session: DBSession, mp4 = false) {
-  return pathJoin(
+async function getRecordingUrls(
+  config: CliArgs,
+  ingest: string,
+  session: DBSession
+): Promise<{ recordingUrl: string; mp4Url: string }> {
+  const asset = await db.asset.getBySessionId(session.id);
+  if (!asset || ["waiting", "processing"].includes(asset.status?.phase)) {
+    // Recording processing in progress
+    return;
```

Comment:
It's probably more future-proof to check for the single phase where we expect a recording to be available
```suggestion
  if (asset?.status?.phase !== "ready") {
    // Recording processing in progress
    return;
```

Reply:
I think we would also need to check `failed`, so it's actually the same.